src/physics/zenith.jl

```diff
@@ -0,0 +1,283 @@
+    return axial_tilt*sin(g-2*(π*convert(NF,equinox)))
+end
+
+"""Coefficients to calculate the solar declination angle δ from
+
+    δ = 0.006918    - 0.399912*cos(g)  + 0.070257*sin(g)
+                    - 0.006758*cos(2g) + 0.000907*sin(2g)
+                    - 0.002697*cos(3g) + 0.001480*sin(3g)
+
+with g the angular fraction of the year in radians. Following Spencer 1971,
+Fourier series representation of the position of the sun. Search 2(5):172.
+$(TYPEDFIELDS)"""
+Base.@kwdef struct SolarDeclination{NF} <: AbstractSolarDeclination{NF}
+    a::NF =   0.006918      # the offset +
+    s1::NF =  0.070257      # s1*sin(g) +
+    c1::NF = -0.399912      # c1*cos(g) +
+    s2::NF =  0.000907      # s2*sin(2g) +
+    c2::NF = -0.006758      # c2*cos(2g) +
+    s3::NF =  0.001480      # s3*sin(3g) +
+    c3::NF = -0.002697      # c3*cos(3g)
+end
+
+"""Generator function pulling the number format NF from a SpectralGrid."""
+SolarDeclination(SG::SpectralGrid;kwargs...) = SolarDeclination{SG.NF}(;kwargs...)
+
+"""
+$(TYPEDSIGNATURES)
+SolarDeclination functor, computing the solar declination angle of
+angular fraction of year g [radians] using the coefficients of the
+SolarDeclination struct."""
+function (SD::SolarDeclination)(g)
+    (;a,s1,s2,s3,c1,c2,c3) = SD
+    sin1g,cos1g = sincos(g)
+    sin2g,cos2g = sincos(2g)
+    sin3g,cos3g = sincos(3g)
+    return a + s1*sin1g + c1*cos1g + s2*sin2g + c2*cos2g + s3*sin3g + c3*cos3g
+end
+
+function Base.show(io::IO,L::AbstractSolarDeclination)
+    println(io,"$(typeof(L)) <: AbstractSolarDeclination")
+    keys = propertynames(L)
+    print_fields(io,L,keys)
+end
+
+"""Coefficients for the solar time correction (also called
+Equation of time) which adjusts the solar hour to an oscillation
+of sunrise/set by about +-16min throughout the year."""
+Base.@kwdef struct SolarTimeCorrection{NF} <: AbstractSolarTimeCorrection{NF}
+    a::NF =   0.004297      # the offset +
+    s1::NF = -1.837877      # s1*sin(g) +
+    c1::NF =  0.107029      # c1*cos(g) +
+    s2::NF = -2.340475      # s2*sin(2g) +
+    c2::NF = -0.837378      # c2*cos(2g)
+end
+
+"""
+$(TYPEDSIGNATURES)
+Functor that returns the time correction for a angular
+fraction of the year g [radians], so that g=0 for Jan-01 and g=2π for Dec-31."""
+function (STC::SolarTimeCorrection)(g)
+    (;a,s1,s2,c1,c2) = STC
+    sin1g,cos1g = sincos(g)
+    sin2g,cos2g = sincos(2g)
+    return deg2rad(a + s1*sin1g + c1*cos1g + s2*sin2g + c2*cos2g)
+end
+
+function Base.show(io::IO,L::AbstractSolarTimeCorrection)
+    println(io,"$(typeof(L)) <: AbstractSolarTimeCorrection")
+    keys = propertynames(L)
+    print_fields(io,L,keys)
+end
+
+"""
+$(TYPEDSIGNATURES)
+Chooses from SolarZenith (daily and seasonal cycle), SolarZenithDay (daily cycle only),
+SolarZenithSeason (seasonal cycle only) and SolarZenithConstant (no daily or seasonal cycle)
+given the parameters in model.planet."""
```

Comment:
Solar Zenith constant would be the latitudinal, yearly averaged, incoming solar radiation?